okhttp/src/test/java/io/grpc/transport/okhttp/OkHttpClientTransportTest.java

```diff
@@ -1004,6 +1056,87 @@ public void ping_failsIfTransportFails() throws Exception {
         ((StatusException) callback.failureCause).getStatus().getCode());
   }
 
+  @Test
+  public void writeBeforeConnected() throws Exception {
+    initTransport(3, new ConnectedCallback(true));
+    final String message = "Hello Server";
+    MockStreamListener listener = new MockStreamListener();
+    OkHttpClientStream stream = clientTransport.newStream(method, new Metadata.Headers(), listener);
+    InputStream input = new ByteArrayInputStream(message.getBytes(UTF_8));
+    stream.writeMessage(input);
+    stream.flush();
+    // The message should be queued.
+    verifyNoMoreInteractions(frameWriter);
+
+    connectedCallback.connected();
```

Comment:
I think the name confused you, it is not used to wait until the connection connected.

I renamed it to "allowConnected".

I need to make sure the tested operations (writing messages here) happens before the connection is connected, so I need to block the connecting thread, do the tested operations, then unblock the connecting thread and check.